docs/api.md

```diff
@@ -1569,6 +1571,31 @@ await navigationPromise; // The navigationPromise resolves after navigation has
 
 **NOTE** Usage of the [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) to change the URL is considered a navigation.
 
+#### page.waitForRequest(url, options)
+- `url` <[string]|[RegExp]> A URL to wait for.
+- `options` <[Object]> Optional waiting parameters
+  - `timeout` <[number]> Maximum wait time in milliseconds, defaults to 30 seconds, pass `0` to disable the timeout.
```

Comment:
let's add a test that validates timeout `0`.
The test should do a 50ms delay before responding to a request; this way we'll demonstrate that timeout 0 is not handled literally.

Reply:
Awesome. Thanks for the feedback. I've made some changes and added timeout = 0 test case.